packages/core/src/interactions/packet/acknowledgement.ts

```diff
@@ -1,33 +1,54 @@
 import debug from 'debug'
-import { AcknowledgementMessage } from '../../messages/acknowledgement'
+import { Acknowledgement } from '../../messages/acknowledgement'
+import { PublicKey, Hash, durations } from '@hoprnet/hopr-utils'
+import PeerId from 'peer-id'
+import HoprCoreEthereum from '@hoprnet/hopr-core-ethereum'
+import { Packet } from '../../messages/packet'
 import { PROTOCOL_ACKNOWLEDGEMENT } from '../../constants'
 import { HoprDB } from '@hoprnet/hopr-utils'
 const log = debug('hopr-core:acknowledgement')
 
+const ACKNOWLEDGEMENT_TIMEOUT = durations.seconds(2)
+
 export function subscribeToAcknowledgements(
   subscribe: any,
   db: HoprDB,
-  paymentChannels: any,
-  onMessage: (ackMessage: AcknowledgementMessage) => void
+  chain: HoprCoreEthereum,
+  pubKey: PeerId,
+  onMessage: (ackMessage: Acknowledgement) => void
 ) {
-  subscribe(PROTOCOL_ACKNOWLEDGEMENT, async function (msg: Uint8Array) {
-    const ackMsg = AcknowledgementMessage.deserialize(msg)
-    let unacknowledgedTicket = await db.getUnacknowledgedTicketsByKey(ackMsg.getHashedKey())
+  subscribe(PROTOCOL_ACKNOWLEDGEMENT, async function (msg: Uint8Array, remotePeer: PeerId) {
+    const ackMsg = Acknowledgement.deserialize(msg, pubKey, remotePeer)
+    let unacknowledgedTicket = await db.getUnacknowledgedTicketsByKey(ackMsg.ackChallenge)
+
     if (!unacknowledgedTicket) {
       // Could be dummy, could be error.
       log('dropping unknown ticket')
-      return await db.deleteTicket(ackMsg.getHashedKey())
+      return await db.deleteTicket(ackMsg.ackChallenge)
     }
 
-    const acknowledgement = await paymentChannels.account.acknowledge(unacknowledgedTicket, ackMsg.getHashedKey())
+    const channel = chain.getChannel(new PublicKey(pubKey.pubKey.marshal()), new PublicKey(remotePeer.pubKey.marshal()))
 
-    if (acknowledgement === null) {
+    const ackedTicket = await channel.acknowledge(unacknowledgedTicket, new Hash(ackMsg.ackKeyShare))
+
+    if (ackedTicket === null) {
       log(`Got a ticket that is not a win. Dropping ticket.`)
-      await db.deleteTicket(ackMsg.getHashedKey())
+      await db.deleteTicket(ackMsg.ackChallenge)
     } else {
       log(`Storing winning ticket`)
-      await db.replaceTicketWithAcknowledgement(ackMsg.getHashedKey(), acknowledgement)
+      await db.replaceTicketWithAcknowledgement(ackMsg.ackChallenge, ackedTicket)
     }
+
     onMessage(ackMsg)
   })
 }
+
+export function sendAcknowledgement(packet: Packet, destination: PeerId, sendMessage: any, privKey: PeerId): void {
```

Comment:
why did you take this out of forward.ts?

Reply:
... because it is part of the PoR scheme and PoR should be decapsulated from packet transformation